pkg/controller/node/rate_limited_queue.go

```diff
@@ -216,3 +219,24 @@ func (q *RateLimitedTimedQueue) Remove(value string) bool {
 func (q *RateLimitedTimedQueue) Clear() {
 	q.queue.Clear()
 }
+
+// SwapLimiter safely swaps current limiter for this queue with the passed one if capacities or qps's differ.
+func (q *RateLimitedTimedQueue) SwapLimiter(limiter flowcontrol.RateLimiter) {
+	if q.limiter.Capacity() == limiter.Capacity() && q.limiter.QPS() == limiter.QPS() {
```

Comment:
This is accessing q.limitted, so also should be done under lock (potentially a different thread can SwapLimitters at the same time.